src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,86 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import { MouseEventHandler } from 'react';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+  onPreviousPage: MouseEventHandler<HTMLAnchorElement>
+  onNextPage: MouseEventHandler<HTMLAnchorElement>
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+  onPreviousPage,
+  onNextPage,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+
+  const handleSetPage = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+
+    const page = Number(event.currentTarget.dataset.currentPage);
+
+    if (page === currentPage) {
+      return;
+    }
+
+    onPageChange(page);
+  };
+
+  return (
+    <ul className="pagination">
+      <li
+        className={classNames(
+          'page-item', { disabled: currentPage === 1 },
+        )}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={onPreviousPage}
```

Comment:
```suggestion
          onClick={handlePreviousPage} // add new handler inside current component
```

You don't need this handler, cause you have `currentPage` value in props and `onPageChange` also. You can use only them to pass new value to parent component.